ui/components/institutional/qr-code-modal/qr-code-modal.tsx

```diff
@@ -204,57 +209,49 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
-        {qrCodeValue ? (
-          <Box
-            data-testid="qr-code-visible"
-            style={{
-              padding: 20,
-              backgroundColor: 'var(--qr-code-white-background)',
-              display: 'flex',
-              alignItems: 'center',
-              flexDirection: 'column',
-            }}
+
+        <ModalBody>
+          {error && <Text color={TextColor.errorDefault}>{error}</Text>}
+          {qrCodeValue ? (
+            <Box
+              data-testid="qr-code-visible"
+              paddingTop={4}
+              paddingBottom={4}
+              display={Display.Flex}
+              alignItems={AlignItems.center}
+              flexDirection={FlexDirection.Column}
+              style={{
+                backgroundColor: 'var(--qr-code-white-background)',
+              }}
+            >
+              <QRCode value={qrCodeValue} size={270} />
+            </Box>
+          ) : (
+            <Spinner color="var(--color-icon-muted)" />
+          )}
+          <Text
+            paddingBottom={4}
+            color={TextColor.textDefault}
+            textAlign={TextAlign.Center}
+            fontWeight={FontWeight.Bold}
+          >
+            {t('custodianQRCodeScan', [custodianName])}
+          </Text>
+          <Text
+            color={TextColor.textDefault}
+            variant={TextVariant.bodySm}
+            textAlign={TextAlign.Center}
           >
-            <QRCode value={qrCodeValue} size={270} />
-          </Box>
-        ) : (
-          <Spinner color="var(--color-warning-default)" />
-        )}
-        <Text
-          as="p"
-          paddingRight={10}
-          paddingLeft={10}
-          paddingBottom={4}
-          color={TextColor.textDefault}
-          textAlign={TextAlign.Center}
-          fontWeight={FontWeight.Bold}
-        >
-          {t('custodianQRCodeScan', [custodianName])}
-        </Text>
-        <Text
-          as="p"
-          paddingRight={10}
-          paddingLeft={10}
-          paddingBottom={4}
-          color={TextColor.textDefault}
-          variant={TextVariant.bodySm}
-          textAlign={TextAlign.Center}
-        >
-          {t('custodianQRCodeScanDescription', [custodianLink])}
-        </Text>
-        <ButtonLink
-          paddingRight={10}
-          paddingLeft={10}
-          paddingBottom={4}
-          variant={TextVariant.bodySm}
-          textAlign={TextAlign.Center}
-          color={TextColor.primaryDefault}
-          onClick={handleClose}
-          data-testid="cancel-btn"
-        >
-          {t('cancel')}
-        </ButtonLink>
+            {t('custodianQRCodeScanDescription', [custodianLink])}
+          </Text>
+        </ModalBody>
+        <ModalFooter
+          onCancel={handleClose}
+          cancelButtonProps={{
+            children: t('cancel'),
+            'data-testid': 'cancel-btn',
+          }}
+        />
         <span className="hidden" data-value={qrCodeValue} />
```

Comment:
### After



https://github.com/MetaMask/metamask-extension/assets/8112138/3b8ffbbb-3e4a-45d3-9596-876ce1e3c485

Reply:
I'd split this out to a separate PR since it may cause some regressions @georgewrmarshall

Reply:
Good suggestion! Done https://github.com/MetaMask/metamask-extension/pull/25011